pkg/trace/agent/obfuscate.go

```diff
@@ -176,3 +176,62 @@ func (cco *ccObfuscator) MetaHook(k, v string) (newval string) {
 	}
 	return v
 }
+
+// MetaStructHook checks the tag with the given key and val and returns the final
+// value to be assigned to this tag.
+func (cco *ccObfuscator) MetaStructHook(k string, v []byte) (newval []byte) {
+	switch k {
+	case "appsec":
+		var obfuscated bool
+		var appsec pb.AppSecStruct
+		_, err := appsec.UnmarshalMsg(v)
+		if err != nil {
+			// Not an appsec struct, ignore the value and log an error
+			log.Errorf("Error obfuscating appsec struct: %v", err)
+			return v
+		}
+		triggers := appsec.GetTriggers()
+		if triggers == nil {
+			return v
+		}
+		for _, trigger := range triggers {
+			matches := trigger.GetRuleMatches()
+			if matches == nil {
+				continue
+			}
+			for _, rulematch := range matches {
+				parameters := rulematch.GetParameters()
+				if parameters == nil {
+					continue
+				}
+				for _, parameter := range parameters {
+					if obfuscate.IsCardNumber(parameter.Value, cco.luhn) {
+						parameter.Value = "?"
+						obfuscated = true
+					}
+					if parameter.Highlight == nil {
+						continue
+					}
+					for j, highlight := range parameter.Highlight {
+						if obfuscate.IsCardNumber(highlight, cco.luhn) {
+							parameter.Highlight[j] = "?"
+							obfuscated = true
+						}
+					}
+				}
+			}
+		}
+		if obfuscated {
+			newval, err := appsec.MarshalMsg(nil)
+			if err != nil {
+				log.Errorf("Error replacing obfuscated appsec struct: %v", err)
+				return v
+			}
+			return newval
+		}
+	default:
+		// Do not obfuscate unknown structures
+		log.Debugf("Obfuscating unknown meta struct is not supported for key: %v", k)
```

Comment:
Do we really need this? Seems like it could get spammy at some point. I'd leave it out.

Reply:
Can get spammy in debug mode, but shouldn't write anything on the output by default. The reason for this message is that if anyone adds a new entry to the `meta_struct` field, it should also add support for it in the obfuscator, so it would be a way to tell them during development at least.

Reply:
Cool. That makes sense. Resolving.